determined-ai/determined
# chore: configure a training port offset

## Description

This change allows an instance of Determined master to be configured to not use specific ports so that a determined master can avoid port conflicts with other master instances (or other software).





## Test Plan

Tested manually:

1. Update devcluster.yaml with:
```
        reserved_ports:
          - 12350
          - 12351
```

2. Run an experiment.

3. Inspect the master log to verify ports allocated as expected:

`master: DEBU[2023-10-11T16:41:30-04:00] DTRAIN_SSH_PORT port : 12352                  experiment-id=108 job-id=9b008a87-6514-4423-bbdc-df3f7dd42d40 task-id=108.2426aa5d-9452-445f-a395-1d807b4e31da task-type=TRIAL trial-id=108 trial-run-id=1
`




## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
